### PR TITLE
fix interaction between include / sparse fields

### PR DESCRIPTION
**Pull request by shadanan**

### CODE REVIEW

File: lib/datastores/jsonapidatastore.js [hunk 1]
```diff
@@ -1032,26 +1068,13 @@ class JsonApiDatastore extends Datastore {
...
-          fieldParams.push(inverse);
-        }
-      });
-
-      linkNames.forEach((linkName) => {
-        fieldParams.push(linkName);
-      });
-      urlParams['fields[' + fieldsModel + ']'] = fieldParams.join(',');
+      if (options && options.include instanceof Object) {
+        Object.keys(this._defaultSparseFields).forEach((modelName) => {
```
Comment (clayreimann) [inline lib/datastores/jsonapidatastore.js:1072]: What if they specify a fieldset too?

File: spec/stores/jsonapidatastore.spec.js [hunk 2]
```diff
@@ -351,7 +351,7 @@ describe('JsonApiDatastore', function() {
         var authors = results['included'];
         expect(authors).to.be.not.empty;
         for (i = 0; i < authors.length; i++) {
-          expect(authorIds).to.have.property(authors[i]['id'])
+          expect(authorIds).to.have.property(authors[i]['name'])
```
Comment (clayreimann) [inline spec/stores/jsonapidatastore.spec.js]: `authors[i].name`

File: lib/datastores/jsonapidatastore.js (hunk 1 above)
Comment (shadanan) [inline lib/datastores/jsonapidatastore.js:1072]: That's handled by the original if

Comment (shadanan) [inline lib/datastores/jsonapidatastore.js:1072]: Added explanations to README as per discussion offline.

File: README.md [hunk 3]
```diff
@@ -156,22 +156,36 @@ elide.find('book', {include: ['authors']})
...
-`[`title`, `authors`, `id`]`, the below query can be used to return the books with title Harry Potter
+**Note:** If you specify a fields option, it overrides the fields for **all models**.
+What this means is that if you query for books, include authors and ask for title and 
+authors of books, you will not get any fields back for authors. In addition to
+title and authors of books, you will also have to include name of authors, as follows:
+
+```javascript
+elide.find('book', {include: ['authors'], fields: {book: ['title', 'authors'], author: ['name']}})
+  .then((results) => {
+    console.log(results.data); // only author information will be available
```
Comment (clayreimann) [inline README.md]: `// only book.title, book.authors, author.name will be available`

**Comment by clayreimann** [comment]
:+1:

**Comment by yahoocla** [comment]
CLA is valid!

**Pull request merged**